central/reports/service/v2/service_impl.go

```diff
@@ -394,3 +395,50 @@ func (s *serviceImpl) DownloadReport(ctx context.Context, req *apiV2.DownloadRep
 
 	return &apiV2.DownloadReportResponse{Data: buf.Bytes()}, nil
 }
+
+func (s *serviceImpl) DeleteReport(ctx context.Context, req *apiV2.DeleteReportRequest) (*apiV2.Empty, error) {
+	if req == nil || req.GetId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Empty request or report job id")
+	}
+
+	slimUser := authn.UserFromContext(ctx)
+	if slimUser == nil {
+		return nil, errors.New("Could not determine user identity from provided context")
+	}
+
+	rep, found, err := s.snapshotDatastore.Get(ctx, req.GetId())
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error finding report snapshot with job ID %q.", req.GetId())
+	}
+
+	if !found {
+		return nil, errors.Wrapf(errox.NotFound, "Error finding report snapshot with job ID '%q'.", req.GetId())
+	}
+
+	if slimUser.GetId() != rep.GetRequester().GetId() {
+		return nil, errors.Wrap(errox.NotAuthorized, "Report cannot be deleted by a user who did not request the report.")
+	}
+
+	status := rep.GetReportStatus()
+	if status.GetReportNotificationMethod() != storage.ReportStatus_DOWNLOAD {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Report job id %q did not generate a downloadable report and hence no report to delete.", req.GetId())
+	}
```

Comment:
Seems like this is reusable piece of code and can be pulled out into a separate function.

Reply:
Yes, slight difference in the wording. I can reorg it later.